spring-batch-docs/asciidoc/whatsnew.adoc

```diff
@@ -4,69 +4,151 @@
+            jobLauncherTestUtils.getUniqueJobParameters();
+
+      // when
+      JobExecution jobExecution =
+            jobLauncherTestUtils.launchJob(jobParameters);
+
+      // then
+      Assert.assertEquals(ExitStatus.COMPLETED,
+                          jobExecution.getExitStatus());
+   }
+
+}
+----
+
+For more details about this new annotation, see the
+<<testing.adoc#creatingUnitTestClass,Unit Testing>> chapter.
+
+[[whatsNewIntegration]]
+=== `@EnableBatchIntegration` Annotation
+
+Setting up a remote chunking job requires the definition of a number of beans:
+
+* A connection factory to acquire connections from the messaging middleware (JMS, AMQP, and others)
+* A `MessagingTemplate` to send requests from the master to the workers and back again
+* An input channel and an output channel for Spring Integration to get messages from the messaging middleware
+* A special item writer (`ChunkMessageChannelItemWriter`) on the master side that knows how to send chunks of data to workers for processing and writing
+* A message listener (`ChunkProcessorChunkHandler`) on the worker side to receive data from the master
+
+This can be a bit daunting at first glance. This release introduces a new annotation
+named `@EnableBatchIntegration` as well as new APIs (`RemoteChunkingMasterStepBuilder`
+and `RemoteChunkingWorkerBuilder`) to simplify the configuration. The following
+example shows how to use the new annotation and APIs:
+
+[source, java]
+----
+@Configuration
+@EnableBatchProcessing
+@EnableBatchIntegration
+public class RemoteChunkingAppConfig {
+
+   @Autowired
+   private RemoteChunkingMasterStepBuilderFactory masterStepBuilderFactory;
+
+   @Autowired
+   private RemoteChunkingWorkerBuilder workerBuilder;
+
+   @Bean
+   public TaskletStep masterStep() {
+         return this.masterStepBuilderFactory
+         	        .get("masterStep")
+         	        .chunk(100)
+         	        .reader(itemReader())
+         	        .outputChannel(outgoingRequestsToWorkers())
+         	        .inputChannel(incomingRepliesFromWorkers())
+         	        .build();
+   }
+
+   @Bean
+   public IntegrationFlow worker() {
+         return this.workerBuilder
+         	        .itemProcessor(itemProcessor())
+         	        .itemWriter(itemWriter())
+         	        .inputChannel(incomingRequestsFromMaster())
+         	        .outputChannel(outgoingRepliesToMaster())
+         	        .build();
+   }
+}
+----
+
+This new annotation and builders take care of the heavy lifting of configuring
+infrastructure beans. You can now easily configure a master step as well as
+a Spring Integration flow on the worker side. You can find a remote chunking sample
+that uses these new APIs in the
+link:$$https://github.com/spring-projects/spring-batch/tree/master/spring-batch-samples#remote-chunking-sample$$[samples module]
+as well as more details in the <<spring-batch-integration.adoc#remote-chunking,Spring Batch Integration>> chapter.
```

Comment:
Somewhere in this section we should note that the channel/adapter configuration is still needed and not in this listing.  Either via an `@Import(RabbitChannelIntegration.class)` or a note or something.

Reply:
In the [remote chunking section](https://github.com/spring-projects/spring-batch/blob/master/spring-batch-docs/asciidoc/spring-batch-integration.adoc#remote-chunking), I used a comment to mention that, see `// Middleware beans setup omitted`. Would this be ok if I add it to the sample of the "what's new" section?

Reply:
That's fine by me.  I just think we can't imply that the configuration of those other pieces isn't needed.

Reply:
ok, fixed.